wandb/lib/ipython.py

```diff
@@ -1,8 +1,12 @@
 def _get_python_type():
     try:
-        if 'terminal' in get_ipython().__module__:
-            return 'ipython'
-        else:
-            return 'jupyter'
-    except (NameError, AttributeError):
+        from IPython import get_ipython
+    except ImportError:
         return "python"
+
+    if get_ipython() is None:
```

Comment:
Don't you mean `get_ipython` ?

Reply:
Nope. We are testing the return value of the function.

Reply:
But `get_ipython` can be None in the `ImportError` case...

Reply:
In the `ImportError` case we return, so rest of the code is not executed.

Reply:
Duh, cool 👍